src/codegate/pipeline/base.py

```diff
@@ -231,8 +231,45 @@ def get_last_user_message(
             return None
         for i in reversed(range(len(request["messages"]))):
             if request["messages"][i]["role"] == "user":
-                content = request["messages"][i]["content"]
-                return content, i
+                content = request["messages"][i]["content"]  # type: ignore
+                return str(content), i
+
+        return None
+
+    @staticmethod
+    def get_last_user_message_block(
+        request: ChatCompletionRequest,
+    ) -> Optional[str]:
+        """
+        Get the last block of consecutive 'user' messages from the request.
+
+        Args:
+            request (ChatCompletionRequest): The chat completion request to process
+
+        Returns:
+            Optional[str]: A string containing all consecutive user messages in the
+                        last user message block, separated by newlines, or None if
+                        no user message block is found.
+        """
+        if request.get("messages") is None:
+            return None
+
+        user_messages = []
+        messages = request["messages"]
+
+        # Iterate in reverse to find the last block of consecutive 'user' messages
+        for i in reversed(range(len(messages))):
+            if messages[i]["role"] == "user" or messages[i]["role"] == "assistant":
```

Comment:
Why do we include the assistant messages? This reads to me as if we would get all the user messages until the first system message?

Maybe it would be nice to have some tests so that the purpose is easier to see through them?

Reply:
when testing deeper with aider, i saw that some of the blocks were composed by this pattern:

system
user
assistant
user

so we need to extract all this block. I am fixing tests now and i can add a test with the aider pattern for it

Reply:
> when testing deeper with aider, i saw that some of the blocks were composed by this pattern:
> 
> system user assistant user
> 
> so we need to extract all this block. I am fixing tests now and i can add a test with the aider pattern for it. I also tested against the other tools and does not affect, as they just have a single user message

I'm more concerned if we don't break any other extractions because other assistants might just send `(assistant,user)` in a single turn of a conversation..

Reply:
i tested other tools and they behave properly, but we should be vigilant, yes